src/infoview.ts

```diff
@@ -65,30 +66,31 @@ export class InfoProvider implements TextDocumentContentProvider, Disposable {
             }
             ` +
             workspace.getConfiguration('lean').get('infoViewStyle');
-        this.updatePosition();
 
         this.subscriptions.push(
             this.server.allMessages.on(() => {
                 if (this.updateMessages()) this.fire();
             }),
             this.server.statusChanged.on(() => {
-                this.updateGoal().then((changed) => { if (changed && this.displayGoal()) this.fire(); })
+                if (this.displayMode == DisplayMode.OnlyState) {
+                    this.updateGoal().then((changed) => { if (changed) this.fire(); })
+                }
             }),
-            window.onDidChangeTextEditorSelection(() => this.updatePosition()),
+            window.onDidChangeTextEditorSelection(() => this.updatePosition(false)),
             commands.registerCommand('_lean.revealPosition', this.revealEditorPosition),
             commands.registerCommand('_lean.hoverPosition', (u, l, c) => { this.hoverEditorPosition(u, l, c); }),
             commands.registerCommand('_lean.stopHover', () => { this.stopHover() }),
-            commands.registerCommand('lean.infoView.toggleMode', (editor) => { 
-                this.toggleMode();
+            commands.registerTextEditorCommand('lean.displayGoal', (editor) => {
+                this.setMode(DisplayMode.OnlyState);
+                this.openPreview(editor);
             }),
-            commands.registerCommand('lean.infoView.startStop', (editor) => { 
-                this.startStop();
+            commands.registerTextEditorCommand('lean.displayList', (editor) => {
+                this.setMode(DisplayMode.AllMessage);
+                this.openPreview(editor);
             }),
-            commands.registerTextEditorCommand('lean.infoView', (editor) => {
-                let column = editor.viewColumn + 1;
-                if (column == 4) column = ViewColumn.Three;
-                commands.executeCommand('vscode.previewHtml', this.leanGoalsUri, column,
-                    "Lean Messages");
+            commands.registerTextEditorCommand('lean.stopUpdating', (editor) => {
+                this.stopUpdating();
```

Comment:
Should we make this a toggle?  Right now the button is pretty useless after you've pressed it once.